### PR TITLE
Alter Address from string to object.

### PR DESCRIPTION
**Pull request by bryanjnelson**
Links to [0511](https://seed-company-squad.monday.com/boards/3451697530/pulses/4788460176)

┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/4960805558) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by atGit2021** [comment]
@bryanjnelson I squashed a lot of garbage commits in order to clean up and make the changes more readable. All the Tests pass now, but the only piece of functionality that remains outstanding is the "list" partners/organizations endpoint which doesn't seem to hydrate in any data.  I'm not sure if the issues resides inside of the loader decorator or somewhere else.  I will need your help or Carson's help to figure out why it hydrates correctly in other calls, but not in the partners/organizations list endpoint.

**Comment by atGit2021** [comment]
> @bryanjnelson I squashed a lot of garbage commits in order to clean up and make the changes more readable. All the Tests pass now, but the only piece of functionality that remains outstanding is the "list" partners/organizations endpoint which doesn't seem to hydrate in any data. I'm not sure if the issues resides inside of the loader decorator or somewhere else. I will need your help or Carson's help to figure out why it hydrates correctly in other calls, but not in the partners/organizations list endpoint.

@bryanjnelson,  I was able to get list to work.  Now it's out putting all the organizations and partners!

**Comment by atGit2021** [comment]
@bryanjnelson, I'm not able to add you as a reviewer, but just fyi, it's not ready for review.

File: src/components/organization/organization.service.ts [hunk 1]
```diff
@@ -95,15 +96,39 @@ export class OrganizationService {
...
+    if (!isAddressSame) {
+      changesToVerify = {
+        ...orgChanges,
+        address: address,
+      };
+    } else {
+      changesToVerify = {
+        ...orgChanges,
+      };
+    }
```
Comment (bryanjnelson) [inline src/components/organization/organization.service.ts:117]: Might this be more concise as a ternary?

File: src/components/partner/partner.service.ts [hunk 2]
```diff
@@ -119,15 +121,35 @@ export class PartnerService {
...
+    if (!isAddressSame) {
+      changesToVerify = {
+        ...changes,
+        address: address,
+      };
+    } else {
+      changesToVerify = {
+        ...changes,
+      };
+    }
```
Comment (bryanjnelson) [inline src/components/partner/partner.service.ts:139]: Same as above?

**Comment by bryanjnelson** [comment]
LGTM @atGit2021!  Over to you @CarsonF.

File: src/components/organization/organization.service.ts [hunk 3]
```diff
@@ -95,15 +96,37 @@ export class OrganizationService {
...
-      .verifyChanges(changes);
+    const isAddressSame = isAddressEqual(address, organization.address);
+    const changesToVerify = !isAddressSame
+      ? {
+          ...orgChanges,
+          address: address,
+        }
+      : {
+          ...orgChanges,
+        };
```
Comment (CarsonF) [inline src/components/organization/organization.service.ts:114]:
```suggestion
    const changes = {
      ...this.repo.getActualChanges(organization, input),
      address: ifDiff(compareNullable(isAddressEqual)(input.address, organization.address),
    };
```

File: src/components/organization/organization.service.ts [hunk 4]
```diff
@@ -95,15 +96,37 @@ export class OrganizationService {
...
+
+    if (Object.keys(orgChanges).length > 0 || 'address' in changesToVerify) {
```
Comment (CarsonF) [inline src/components/organization/organization.service.ts:116]:
```suggestion
```

File: src/components/organization/organization.service.ts [hunk 5]
```diff
@@ -95,15 +96,37 @@ export class OrganizationService {
...
+      this.privileges
+        .for(session, Organization, organization)
+        .verifyChanges(changesToVerify);
+    }
+
+    if (Object.keys(orgChanges).length > 0) {
```
Comment (CarsonF) [inline src/components/organization/organization.service.ts:122]:
```suggestion
```

File: src/components/organization/organization.service.ts [hunk 6]
```diff
@@ -95,15 +96,37 @@ export class OrganizationService {
...
+      await this.repo.updateProperties(organization, orgChanges);
+    }
+    if (address && 'address' in changesToVerify) {
```
Comment (CarsonF) [inline src/components/organization/organization.service.ts:125]:
```suggestion
    if (changes.address !== undefined) {
```

File: src/components/organization/organization.repository.ts [hunk 7]
```diff
@@ -26,20 +29,57 @@ export class OrganizationRepository extends DtoRepository<
...
+      })
       .return<{ id: ID }>('node.id as id');
-
     return await query.first();
   }
 
+  async updateOrgAddress(
+    id: ID,
+    session: Session,
+    addressToUpdate: MailingAddress,
```
Comment (CarsonF) [inline src/components/organization/organization.repository.ts:62]:
```suggestion
    addressToUpdate: MailingAddress | null,
```

**Review by CarsonF — CHANGES_REQUESTED** [review]
Mostly just:
- clean up from resource -> value object stuff
- allowing address to be set to null

File: src/components/mailing-address/dto/mailing-address.dto.ts
[the remark this answers is not in the document]
Comment (atGit2021) [reply src/components/mailing-address/dto/mailing-address.dto.ts:1]: moved the mailing address folder to /common

**Comment by sethmcknight** [comment]
This lived to the end of its lifetime and died

**Pull request closed without merging**